### PR TITLE
[SDKS-6544] Add new config localhostRefreshEnabled

### PR DESCRIPTION
**Pull request by nmayorsplit**
# GO SPLIT COMMONS

## What did you accomplish?
- New feature: Add new logic for new localhostRefreshEnabled in local.go.

### CODE REVIEW

**Review by sanzmauro — APPROVED** [review]

File: synchronizer/local.go [hunk 1]
```diff
@@ -61,21 +64,25 @@ func (s *Local) SyncAll() error {
 
 // StartPeriodicFetching starts periodic fetchers tasks
 func (s *Local) StartPeriodicFetching() {
-	if s.splitTasks.SplitSyncTask != nil {
-		s.splitTasks.SplitSyncTask.Start()
-	}
-	if s.splitTasks.SegmentSyncTask != nil {
-		s.splitTasks.SegmentSyncTask.Start()
+	if s.refreshEnabled {
```
Comment (mmelograno) [inline synchronizer/local.go:67]: I don't think that you need to check for this if the tasks are nil

File: synchronizer/local.go [hunk 2]
```diff
@@ -61,21 +64,25 @@ func (s *Local) SyncAll() error {
...
 }
 
 // StopPeriodicFetching stops periodic fetchers tasks
 func (s *Local) StopPeriodicFetching() {
-	if s.splitTasks.SplitSyncTask != nil {
-		s.splitTasks.SplitSyncTask.Stop(false)
-	}
-	if s.splitTasks.SegmentSyncTask != nil {
-		s.splitTasks.SegmentSyncTask.Stop(true)
+	if s.refreshEnabled {
```
Comment (mmelograno) [inline synchronizer/local.go:79]: I don't think that you need to check for this if the tasks are nil

**Review by mmelograno — CHANGES_REQUESTED** [review]

**Review by mmelograno — CHANGES_REQUESTED** [review]
You should check create or not the SyncTasks here
```	
splitTasks := SplitTasks{
		SplitSyncTask: tasks.NewFetchSplitsTask(workers.SplitFetcher, cfg.SplitPeriod, logger),
	}
	if cfg.SegmentDirectory != "" {
		workers.SegmentFetcher = segment.NewSegmentFetcher(splitStorage, segmentStorage, splitAPI.SegmentFetcher, logger, runtimeTelemetry, hcMonitor)
		splitTasks.SegmentSyncTask = tasks.NewFetchSegmentsTask(workers.SegmentFetcher, cfg.SegmentPeriod, cfg.SegmentWorkers, cfg.QueueSize, logger)
	}
	```
	if not where are you controlling creating or not it?

File: synchronizer/local.go (hunk 1 above)
Comment (nmayorsplit) [reply synchronizer/local.go]: I moved the logic to the NewLocal.

File: synchronizer/local.go (hunk 2 above)
Comment (nmayorsplit) [reply synchronizer/local.go]: I moved the logic to the NewLocal.

**Review by mmelograno — APPROVED** [review]

**Pull request merged**